### ISSUE TITLE
Uncaught (in promise) TypeError: this.gl.setCompatibleXRDevice is not a function

### ISSUE THREAD

**Issue by MannyMeadows** [googlecodelabs/ar-with-webxr#18]
Having successfully run the codelab examples yesterday, I've found that after today's Canary update I'm getting the above error on my S8.

The result is that the grey background and GIF remains while the camera feed is missing.

**Comment by DoDecaDence** [comment]
The` WebGLRenderingContext` appears to have changed.
The method `setCompatibleXRDevice` was used for checking device compatibility. The WebXR Device API draft [has been updated just recently](https://immersive-web.github.io/webxr/), so I assume a structural change has to do with it.

![chrome_2019-01-09_webglrenderingcontext_unknownfunction](https://user-images.githubusercontent.com/7498626/50901275-fcd2fd80-1417-11e9-9bc2-b0f98d4101cc.png)

This happens in line 132 in `app.js`:

`this.gl = this.renderer.getContext();`

`// Ensure that the context we want to write to is compatible`
`// with our XRDevice`
`await **this.gl.setCompatibleXRDevice**(this.session.device);`


Regarding Chromium, there are currently [two issues](https://bugs.chromium.org/p/chromium/issues/list?can=1&q=setCompatibleXRDevice&colspec=ID+Pri+M+Stars+ReleaseBlock+Component+Status+Owner+Summary+OS+Modified&x=m&y=releaseblock&cells=ids) about this function name; but they're both from August last year.

I have yet to find out how to properly get the XR device now. Any ideas?
(My guess would be that it might be done using the XR object instead, which is meant to be the entry point to the API, so this could avoid having to touch the GL context directly.)

The [proposal entry](https://webcache.googleusercontent.com/search?q=cache:2LX7dW7ToL4J:https://immersive-web.github.io/webxr-reference/webxr-device-api/partial_webglcontextattributes.html+&cd=1&hl=de&ct=clnk&gl=de) for this method in the Immersive Web Github Repo is even older.

(It seems to have been like that for a long time now, as those lines of code above have also been used on some of [the older webxr-examples](https://github.com/immersive-web/webxr-samples) from March 2018.)

**Comment by jsantell** [comment]
Looks like this function was changed to [makeXRCompatible](https://immersive-web.github.io/webxr/#dom-webglrenderingcontextbase-makexrcompatible)

**Comment by krzysioks** [comment]
Hi,
after following jsantell's comment I have changed call of setCompatibleXRDevice to makeXRCompatible. This solved the original issue, but new one has arised:

`Uncaught (in promise) TypeError: this.session.requestFrameOfReference is not a function
    at App.onSessionStarted`

In the WebXR Device API draft: [https://immersive-web.github.io/webxr/#dom-webglrenderingcontextbase-makexrcompatible](url) there is no more reference to requestFrameOfReference. Might this function been changed also?

**Comment by jsantell** [comment]
There's been a few changes in Canary so the implementation is inbetween documented specs, so a bit hard to follow unfortunately. Check out these changes we've landed in [model-viewer](https://github.com/GoogleWebComponents/model-viewer), it's mostly cosmetic name changes -- FYI this will be happening over the next few months, so the platform is still quite unstable.

Issues:

* https://github.com/GoogleWebComponents/model-viewer/issues/302
* https://github.com/GoogleWebComponents/model-viewer/pull/292
* https://github.com/GoogleWebComponents/model-viewer/pull/287
